src/contexts/auth/provider.ts

```diff
@@ -1,41 +1,47 @@
-import { onAuthStateChanged } from 'firebase/auth'
-import { useEffect } from 'react'
-import { auth } from 'services/firebase'
-import { getUser } from 'domains/auth/services'
+import { onAuthStateChanged } from 'firebase/auth';
+import { useEffect } from 'react';
+import { auth } from '../../services/firebase';
+import { getUser } from '../../domains/auth/services';
+import { AuthContext } from './context';
+import { useState } from 'react';
+
 type Props = {
-	children?: React.ReactNode
+  children?: React.ReactNode;
 }
 
 export function AuthProvider({ children }: Props) {
-	const [firebaseUser, setFirebaseUser] = useState<FirebaseUser>()
+  const [firebaseUser, setFirebaseUser] = useState<FirebaseUser>()
 const [user, setUser] = useState<User>()
 
-	// тут слухати за змінами користувача.
-	// потрібно додати useEffect, в якому додати використання onAuthStateChange
-	// Почитати можна тут
-
-
-	useEffect(()=>{
-		const unsubscibe = onAuthStateChanged(
-			auth,
-			(changedFirebaseUser) => {
-				if(!changedFirebaseUser) {
-					setFirebaseUser(undefined)
-					setUser(undefined)
-					return
-				}
-				const changedUser = await getUser(changedFirebaseUser.uid)
-				setFirebaseUser(changedFirebaseUser)
-				setUser(changedUser)
-	
-			}
-		)
-		return unsubscibe
-	},[])
-
-	return (
-		<AuthContext.Provider value={contextValue}>
-			{children}
-		</AuthContext.Provider>
-	)
-}
\ No newline at end of file
+  useEffect(() => {
+    const fetchData = async () => {
+      const unsubscribe = onAuthStateChanged(auth, async (changedFirebaseUser) => {
+        if (!changedFirebaseUser) {
+          setFirebaseUser(null);
+          setUser(null);
+          return;
+        }
+
+        try {
+          const changedUser = await getUser(changedFirebaseUser.uid);
+          setFirebaseUser(changedFirebaseUser);
+          setUser(changedUser);
+        } catch (error) {
+          console.error("Error fetching user data:", error);
+        }
+      });
+
+      return () => unsubscribe();
+    };
+
+    fetchData();
```

Comment:
You don't need to add a fetchData wrapper. Please, remove it.
```suggestion
    const unsubscribe = onAuthStateChanged(auth, async (changedFirebaseUser) => {
      if (!changedFirebaseUser) {
        setFirebaseUser(null);
        setUser(null);
        return;
      }

      try {
        const changedUser = await getUser(changedFirebaseUser.uid);
        setFirebaseUser(changedFirebaseUser);
        setUser(changedUser);
      } catch (error) {
        console.error("Error fetching user data:", error);
      }
    });

    return unsubscribe;
```